indico/modules/users/client/js/UserDelete.jsx

```diff
@@ -0,0 +1,204 @@
+        {isButtonDisabled ? (
+          <Button color="red" disabled>
+            <Translate>
+              Yes, I want to delete <Param name="first_name" value={firstName} /> (
+              <Param name="countdown_seconds" value={countdown} />)
+            </Translate>
+          </Button>
+        ) : (
+          <Button color="red" onClick={onDelete} disabled={disabled} loading={inProgress}>
+            <Translate>
+              Yes, I want to delete <Param name="first_name" value={firstName} />
+            </Translate>
+          </Button>
+        )}
+      </Modal.Actions>
+    </>
+  );
+}
+
+UserDeleteDialogBody.propTypes = {
+  firstName: PropTypes.string.isRequired,
+  lastName: PropTypes.string.isRequired,
+  disabled: PropTypes.bool.isRequired,
+  inProgress: PropTypes.bool.isRequired,
+  onDelete: PropTypes.func.isRequired,
+  onClose: PropTypes.func.isRequired,
+};
+
+function UserDelete({userId, firstName, lastName}) {
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [deleting, setDeleting] = useState(false);
+  const isSameUser = userId === null;
+
+  const handleCloseDialog = () => {
+    setIsDialogOpen(false);
+  };
+
+  const handleDelete = async () => {
+    setDeleting(true);
+    let resp;
+    try {
+      resp = await indicoAxios.delete(userDeleteURL({user_id: userId}));
+    } catch (err) {
+      setDeleting(false);
+      handleCloseDialog();
+      handleAxiosError(err);
+      return;
+    }
+    setDeleting(false);
+    handleCloseDialog();
+    location.href = resp.data.redirect;
+  };
+
+  return (
+    <div>
+      {isSameUser ? (
+        <Popup
+          trigger={
+            <span>
+              <Button size="small" color="red" disabled>
+                <Translate>Delete User</Translate>
+              </Button>
+            </span>
+          }
+          size="small"
+          wide
+          content={Translate.string('You cannot delete your own account')}
+          position="bottom center"
+        />
+      ) : (
+        <>
+          <Button
+            size="small"
+            color="red"
+            onClick={() => setIsDialogOpen(true)}
+            disabled={isSameUser}
+          >
+            <Translate>Delete User</Translate>
+          </Button>
+          <Modal size="small" open={isDialogOpen} onClose={handleCloseDialog} closeIcon>
+            <UserDeleteDialogBody
+              firstName={firstName}
+              lastName={lastName}
+              disabled={deleting}
+              inProgress={deleting}
+              onDelete={handleDelete}
+              onClose={handleCloseDialog}
+            />
+          </Modal>
+        </>
+      )}
+    </div>
+  );
+}
+
+UserDelete.propTypes = {
+  userId: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf([null])]),
+  firstName: PropTypes.string.isRequired,
+  lastName: PropTypes.string.isRequired,
+};
+
+window.setupUserDelete = function setupUserDelete(userId, firstName, lastName) {
```

Comment:
I would avoid creating those global functions in new code. Custom Elements are much nicer for this, see e.g. [this](https://github.com/indico/indico-plugins/blob/master/vc_zoom/indico_vc_zoom/client/ind_vc_zoom_join_button.js) for a simple example - ie you'd define the custom element in here, and then in the template you just put this:

```html
<ind-user-delete-button user-id="{{ user.id }}" user-first-name="{{ user.first_name }}" user-last-name="{{ user.last_name }}"></ind-user-delete-button>
```